### ISSUE TITLE
nix-prefetch-url -A wants to unpack everything where executable=true 

### ISSUE THREAD

**Issue by 4z3** [NixOS/nix#1514]
When the argument to `fetchurl` defines `executable = true`, then `outputModeHash` gets [set to `"recursive"`](https://github.com/NixOS/nix/blob/fe97c6989841460efca37f0f3b9b470c98229283/corepkgs/fetchurl.nix#L18). This causes `nix-prefetch-url` to [want to unpack the downloaded file](https://github.com/NixOS/nix/blob/2fd8f8bb99a2832b3684878c020ba47322e79332/src/nix-prefetch-url/nix-prefetch-url.cc#L122-L127).  Other tools, like `nix-build`, aren't affected by this issue.

#### Example

Lets assume we add/overlay following package:

```nix
{ fetchurl }:
fetchurl {
  url = "https://nixos.org/releases/nix/nix-1.11.13/manual.pdf";
  sha256 = "0g8pdb900b3mbq3wxy855qfwggcq6njv48g9fiq0jikbnbdmgclv";
  executable = true;
} 
```

Prefetching it will then fail as follows:

```console
$ nix-prefetch-url -A pkgs.derp
downloading ‘https://nixos.org/releases/nix/nix-1.11.13/manual.pdf’... [0/0 KiB, 0.0 KiB/s]
unpacking...
tar: This does not look like a tar archive
tar: Skipping to next header
tar: Exiting with failure status due to previous errors
error: program ‘tar’ failed with exit code 2
```

With `executable = true` removed, everything works as expected.

**Comment by vcunat** [comment]
Right, b54f447df added a heuristics for the `-A` mode: it unpacks if (and only if) `outputHashMode == "recursive"`.  This case isn't covered.

Also, any `postFetch` commands are ignored, so prefetching won't work for `fetchpatch`, for example.  (Confusingly, it will fetch but won't adjust the patch.)

**Comment by ryantm** [comment]
I ran into this issue with `nixpkgs-update` trying to update fixed-output-derivations that use `postFetch`.
